TicTacToe/TicTacToe/WinConditions.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace TicTacToe {
+
+    public class WinConditions {
+
+        public bool IsWin(string[] gameBoard) {
+            return IsHorizontalWin(gameBoard) || IsVerticalWin(gameBoard) || 
+                IsForwardDiagonalWin(gameBoard) || IsBackwardDiagonalWin(gameBoard);
+        }
+
+        private bool IsHorizontalWin(string[] gameBoard) {
+            int rowLength = (int)Math.Sqrt(gameBoard.Length);
+            List<string> row = new List<string>();
+            for (int i = 0; i <= (gameBoard.Length - rowLength); i += rowLength) {
+                row.Add(gameBoard[i]);
+                for (int j = (i + 1); j <= (i + (rowLength - 1)); j++) {
+                    row.Add(gameBoard[j]);
+                }
+                if (row.All(cell => cell == row.First())) {
+                    return true;
+                } else {
+                    row.Clear();
+                }
+            }
+            return false;
+        }
+
+        private bool IsVerticalWin(string[] gameBoard) {
+            int columnLength = (int) Math.Sqrt(gameBoard.Length);
+            List<string> column = new List<string>();
+            for (int i = 0; i < columnLength; i++) {
+                column.Add(gameBoard[i]);
+                for (int j = (i + columnLength); j <= (i + (columnLength * 2)); j += columnLength) {
+                    column.Add(gameBoard[j]);
+                }
+                if (column.All(cell => cell == column.First())) {
+                    return true;
+                } else {
+                    column.Clear();
```

Comment:
Why is `Clear()` called here and above for the row but not below for the diagonals?

These methods should all look nearly identical before addressing duplication with extraction(s).